source/extensions/transport_sockets/tls/cert_validator/spiffe/spiffe_validator.cc

```diff
@@ -82,46 +82,51 @@ SPIFFEValidator::parseTrustBundles(const std::string& trust_bundle_mapping_str)
+                return false;
               }
               const auto& certs = key->getStringArray("x5c");
               if (!certs.ok() || (*certs).size() == 0) {
-                ENVOY_LOG(error, "missing 'x5c' field found in keys for domain: '{}'", domain_name);
-                return (success = false);
+                parsing_status = absl::InvalidArgumentError(fmt::format(
+                    "missing or empty 'x5c' field found in keys for domain: '{}'", domain_name));
+                return false;
               }
               for (const auto& cert : *certs) {
                 std::string decoded_cert = Envoy::Base64::decode(cert);
                 if (decoded_cert.empty()) {
-                  ENVOY_LOG(error, "Invalid or empty cert decoded in domain '{}'", domain_name);
-                  return (success = false);
+                  parsing_status = absl::InvalidArgumentError(
+                      fmt::format("Failed to create x509 object while loading certs in domain '{}'",
+                                  domain_name));
+                  return false;
                 }
 
                 const unsigned char* cert_data =
                     reinterpret_cast<const unsigned char*>(decoded_cert.data());
                 bssl::UniquePtr<X509> x509(d2i_X509(nullptr, &cert_data, decoded_cert.size()));
                 if (!x509) {
-                  ENVOY_LOG(error,
-                            "Failed to create x509 object while loading certs in domain '{}'",
-                            domain_name);
-                  return (success = false);
+                  parsing_status = absl::InvalidArgumentError(
+                      fmt::format("Invalid x509 object in certs for domain '{}'", domain_name));
+                  return false;
                 }
-                if (X509_STORE_add_cert(spiffe_data->trust_bundle_stores[domain_name].get(),
+                if (X509_STORE_add_cert(spiffe_data->trust_bundle_stores_[domain_name].get(),
                                         x509.get()) != 1) {
-                  ENVOY_LOG(error, "Failed to add x509 object while loading certs for domain '{}'",
-                            domain_name);
-                  return (success = false);
+                  parsing_status = absl::InternalError(
+                      fmt::format("Failed to add x509 object while loading certs for domain '{}'",
+                                  domain_name));
+                  return false;
                 }
-                spiffe_data->ca_certs.push_back(std::move(x509));
+                spiffe_data->ca_certs_.push_back(std::move(x509));
               }
             }
 
             return true;
           });
 
-  if (!status.ok() || !success) {
-    return nullptr;
+  // json_object::iterate seems to always return ok(), so this check is
+  // redundant...
+  if (!status.ok() || !parsing_status.ok()) {
+    return parsing_status;
   }
```

Comment:
I agree the check may be redundant. But there is a two-symbolism and may have potential problem. If the `parsing_status` is ok and `status` is not ok because the implementation change, what will happen?
```suggestion
  // json_object::iterate seems to always return ok(), so this check is
  // redundant...
  RETURN_IF_NOT_OK_REF(status);
  RETURN_IF_NOT_OK_REF(parsing_status);
```